File: frontend/src/api/purchaseOrder.ts
import { http } from "./http";

export interface PurchaseOrderListItem {
  id: number; orderNo: string; applicationNo: string | null; status: string;
  supplierName: string; itemName: string; orderQuantity: string;
  totalAmount: string; currency: string; createdAt: string;
}
export interface PurchaseOrderDetail extends PurchaseOrderListItem {
  shippingMethod: string | null; expectedDeliveryDate: string | null;
  paymentTerms: string | null; itemModel: string | null; unit: string;
  unitPrice: string; taxInclusiveAmount: string; taxRateName: string | null;
}
export interface PurchaseOrderListResponse { total: number; items: PurchaseOrderListItem[]; }
export const purchaseOrderApi = {
  list: (params: { page?: number; size?: number }) =>
    http.get<unknown, { code: number; message: string; data: PurchaseOrderListResponse }>("/purchase-orders", { params }),
  get: (id: number) =>
    http.get<unknown, { code: number; message: string; data: PurchaseOrderDetail }>(`/purchase-orders/${id}`),
};

File: backend/src/main/java/com/example/contractagent/order/PurchaseOrder.java
package com.example.contractagent.order;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@TableName("purchase_order")
public class PurchaseOrder {
    @TableId(type = IdType.AUTO)
    private Long id;
    private String orderNo;
    private Long taskId;
    private Long extractionId;
    private String applicationNo;
    private Long productId;
    private String supplierName;
    private String itemName;
    private String itemModel;
    private String unit;
    private BigDecimal requestedQuantity;
    private BigDecimal inventoryQuantity;
    private BigDecimal orderQuantity;
    private BigDecimal unitPrice;
    private BigDecimal totalAmount;
    private String currency;
    private String taxRateName;
    private LocalDate expectedDeliveryDate;
    private String shippingMethod;
    private String deliveryLocation;
    private String paymentTerms;
    private String inventoryCheckStatus;
    private String status;
    private Long createdBy;
    private LocalDateTime createdAt;
    private LocalDateTime submittedAt;
}
